script/cli/src/libs/explorer.rs

```diff
@@ -8,102 +7,98 @@ use alloy::{
 #[derive(Clone, PartialEq, Eq, Default)]
 pub enum SupportedExplorerType {
     #[default]
+    Manual,
+    EtherscanV2,
     Etherscan,
     Blockscout,
 }
 
 #[derive(Default, Clone)]
-pub struct ExplorerApiLib {
+pub struct Explorer {
     pub name: String,
     pub url: String,
     pub standard: String,
+    pub explorer_type: SupportedExplorerType,
+}
+
+#[derive(Default, Clone)]
+pub struct ExplorerApiLib {
+    pub explorer: Explorer,
     pub api_key: String,
     pub api_url: String,
-    pub explorer_type: SupportedExplorerType,
 }
 
 impl ExplorerApiLib {
     pub fn new(explorer: Explorer, api_key: String) -> Result<Self, Box<dyn std::error::Error>> {
-        if explorer.name.to_lowercase().contains("blockscout") {
+        if explorer.explorer_type == SupportedExplorerType::Blockscout {
```

Comment:
Recommend a `match` statement rather than if tree. better type checking for case coverage